modules/manifold/src/components/ui/segment-filters-control/segment-panel-list.js

```diff
@@ -0,0 +1,64 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import styled, {css} from 'styled-components';
+
+import {FILTER, FIELD} from '../../../constants';
+import SegmentPanel from './segment-panel';
+
+const SegmentPanelListContainer = styled.div`
+  color: #666;
+  font-family: ff-clan-web-pro, 'Helvetica Neue', Helvetica, sans-serif;
+  font-size: 12px;
+
+  ${props =>
+    props.withPadding &&
+    css`
+      background: #fff;
+      padding: 12px;
+    `};
+`;
+
+export default class SegmentPanelList extends PureComponent {
+  static propTypes = {
+    columnDefs: PropTypes.arrayOf(FIELD),
+    segmentFilters: PropTypes.arrayOf(
+      PropTypes.shape({
+        _key: PropTypes.string,
+        _content: PropTypes.arrayOf(FILTER),
+      })
+    ),
+    onUpdateFilterValue: PropTypes.func,
+  };
+
+  static defaultProps = {
+    baseCols: [],
+    columnDefs: [],
+    segmentFilters: [[], []],
+    onUpdateFilterValue: () => {},
+  };
+
+  render() {
+    const {
+      segmentFilters,
+      columnDefs,
+      onRemoveSegment,
+      onUpdateFilterValue,
+    } = this.props;
+
+    return (
+      <SegmentPanelListContainer>
+        {segmentFilters.map((singleSegmentFilters, i) => (
+          <SegmentPanel
+            key={singleSegmentFilters._key}
```

Comment:
Can we just use `i` as the key? React will take care of the prefixes.
From line 36, the default segmentFilters are two empty arrays, `singleSegmentFilters._key` will be undefined, will collide.

Also, if this is the only place we needed `singleSegmentFilters._key`, we may want to remove it completely and pull `_content` one level above.

Reply:
We cannot use `i` as key because we want to achieve the UI effect that, if user deletes `segment 2` which was closed, `segment 3` which was initially open becomes `segment 2`, but remains open. I.e. we cannot use array element index since elements can be deleted but we might want to keep component states.

Reply:
>Also, if this is the only place we needed singleSegmentFilters._key, we may want to remove it completely and pull _content one level above.

Can you elaborate this?

Reply:
> Can you elaborate this?

nvm if we need to keep the key as you explained above. 👍